lib/queues.js

```diff
@@ -163,15 +164,24 @@ class Queue {
   // Push job to queue with the given delay.  Delay can be millisecond,
   // or a string of the form '5s', '2m', etc.
   delayJob(job, duration) {
+    const self = this;
+
     return this.delayJobs([ job ], duration)
-      .then(jobIDs => jobIDs[0]);
+      .then(function(jobIDs) {
+
+        if (process.env.NODE_ENV === 'test')
+          self._delayedJobs.set(jobIDs[0], Date.now() + ms(duration));
```

Comment:
I'm going to go with your suggestion - delayJobs already has the assertion against duration's value  + converts duration value for us. 👍